hebiros_gazebo_plugin/include/hebiros_gazebo_controller.cpp

```diff
@@ -0,0 +1,383 @@
+
+#include <hebiros_gazebo_controller.h>
+
+
+HebirosGazeboController::HebirosGazeboController() {}
+
+HebirosGazeboController::~HebirosGazeboController() {}
+
+//Set defaults settings for a joint once
+void HebirosGazeboController::SetSettings(
+  std::shared_ptr<HebirosGazeboJoint> hebiros_joint) {
+
+  CommandMsg target = hebiros_joint->command_target;
+  int i = hebiros_joint->command_index;
+
+  hebiros_joint->low_pass_alpha = LOW_PASS_ALPHA;
+
+  //Set gear ratio
+  if (gear_ratios.find(hebiros_joint->model_name) != gear_ratios.end()) {
+    hebiros_joint->gear_ratio = gear_ratios[hebiros_joint->model_name];
+  }
+  else {
+    hebiros_joint->gear_ratio = DEFAULT_GEAR_RATIO;
+  }
+
+  //Set control strategy
+  if (i < target.settings.control_strategy.size()) {
+    hebiros_joint->settings.control_strategy = {target.settings.control_strategy[i]};
+  }
+  else {
+    hebiros_joint->settings.control_strategy = {
+      static_cast<char>(DEFAULT_CONTROL_STRATEGY)};
+  }
+
+  SetDefaultGains(hebiros_joint);
+}
+
+//Initialize gains with default values based on model and control strategy
+void HebirosGazeboController::SetDefaultGains(
+  std::shared_ptr<HebirosGazeboJoint> hebiros_joint) {
+  
+  std::string model_name = hebiros_joint->model_name;
+  int control_strategy = hebiros_joint->settings.control_strategy[0];
+
+  if (model_name == "X5_1" && control_strategy == 2) {
```

Comment:
Eventually, it'd be great to support the common XML format in some form here, which could help reduce this lengthy code here.